pandas/core/indexing.py

```diff
@@ -197,6 +198,19 @@ def _get_setitem_indexer(self, key):
     def __setitem__(self, key, value):
         if isinstance(key, tuple):
             key = tuple(com.apply_if_callable(x, self.obj) for x in key)
+            if (
```

Comment:
Maybe this gets simplified if the `hashable` stuff can be removed, but can you add a comment here as to what is going on? Not immediately apparent what all of these conditions are